crates/sui-graphql-rpc/src/client/response.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::{Response, ServerError, Value};
+use hyper::HeaderMap;
+use reqwest::Response as ReqwestResponse;
+use std::{collections::BTreeMap, net::SocketAddr};
+
+use crate::server::version::VERSION_HEADER;
+
+#[derive(Debug)]
+pub struct GraphqlResponse {
+    headers: HeaderMap,
+    remote_address: Option<SocketAddr>,
+    http_version: hyper::Version,
+    status: hyper::StatusCode,
+    full_response: Response,
+}
+
+impl GraphqlResponse {
+    pub async fn from_resp(resp: ReqwestResponse) -> Self {
+        let headers = resp.headers().clone();
+        let remote_address = resp.remote_addr();
+        let http_version = resp.version();
+        let status = resp.status();
+        let full_response: Response = resp.json().await.expect("Failed to parse response");
+
+        Self {
+            headers,
+            remote_address,
+            http_version,
+            status,
+            full_response,
+        }
+    }
+
+    pub fn graphql_version(&self) -> String {
```

Comment:
I'm assuming this code is only intended for tests, given that there are `expect`s and `panic!`s in it?

Reply:
Indeed this client is test only right now and will be extracted out.
Although its a fair enough request to handle errors better. Fixed it.